### PR TITLE
Add EtcdCluster resource examples

### PR DESCRIPTION
**Pull request by kvaps**
following up https://github.com/aenix-io/etcd-operator/issues/109#issuecomment-2032847557

Here are examples for:

- Simple cluster with few replicas
- Cluster with custom image
- Cluster with persistent storage
- Cluster with resources
- Cluster with all field specified

### CODE REVIEW

**Comment by kvaps** [comment]
I’ll personally raise the question again whether we should replace:

`spec.pod` --> `spec.template`
`spec.service` --> `spec.serviceTemplate`
`spec.podDisruptionBudget` --> `spec.podDisruptionBudgetTemplate`

piraeus-operator and elk-operator use it this way

**Review by sircthulhu — APPROVED** [review]
lgtm

I would personally add `etcdcluster-with-default-pdb.yaml`
```yaml
---
apiVersion: etcd.aenix.io/v1alpha1
kind: EtcdCluster
metadata:
  name: test
  namespace: default
spec:
  replicas: 3
  podDisruptionBudget: {}
```

**Comment by kvaps** [comment]
PR rebased after [internal discussion](https://t.me/etcd_operator/5601) to use Template suffixes for configuring Kubernetes entities.

Another advantage of using the Template suffixes is that it serves as a contract to adhere to the Kubernetes specifications.

What we have in storage and options is validated and processed by our logic, and the specifications there are strictly our own. However, what is in *Template follows the Kubernetes specifications and is intuitively understandable.

**Review by sergeyshevch — APPROVED** [review]

File: examples/manifests/etcdcluster-full.yaml
Comment (lllamnyp) [inline examples/manifests/etcdcluster-full.yaml:1]: In
```yaml
storage:
  emptyDir: {}
  volumeClaimTemplate: # ...
```
how do we know if an emptydir or a PVC will be used?

**Review by hiddenmarten — APPROVED** [review]

File: examples/manifests/etcdcluster-full.yaml
Comment (sircthulhu) [reply examples/manifests/etcdcluster-full.yaml:1]: Currently storage spec is the following:
```go
type StorageSpec struct {
	// EmptyDirVolumeSource to be used by the StatefulSets. If specified, used in place of any volumeClaimTemplate. More
	// info: https://kubernetes.io/docs/concepts/storage/volumes/#emptydir
	// +optional
	EmptyDir *corev1.EmptyDirVolumeSource `json:"emptyDir,omitempty"`
	// A PVC spec to be used by the StatefulSets.
	// +optional
	VolumeClaimTemplate EmbeddedPersistentVolumeClaim `json:"volumeClaimTemplate,omitempty"`
}
```

if emptyDir is null, then PVC. If emptyDir is not null, then emptyDir

**Review by sircthulhu — APPROVED** [review]

File: examples/manifests/etcdcluster-full.yaml
Comment (lllamnyp) [reply examples/manifests/etcdcluster-full.yaml:1]: So the manifest in the example is not really valid, as both `emptyDir` and `volumeClaimTemplate` are not null (emptyDir is an empty non-nil map)?

Comment (kvaps) [reply examples/manifests/etcdcluster-full.yaml:1]: yep, it should be OneOf

**Pull request merged**